### ISSUE TITLE
Microbenchmarks hosting tests can't find httpapi.dll in OSX and Linux

### ISSUE THREAD

**Issue by DavidObando** [aspnet/Performance#61]
**This issue requires more investigation.**

The issue happens while executing the tests as part of the `build.sh` script. It's not confirmed whether this also affects Linux or not.

The issue is repeated for a large number of tests. The first one to blow up like this is:
```
MainToConfigureOverhead [Variation: WebListener] [Stage: Warmup] [Iteration: 1/1]
```

Here's the console output:
```
Published to test/Microbenchmarks.Tests/obj/testPublish-dnx451
Published 1/1 projects successfully
info: Copying HTML.xslt
info: Copying NUnitXml.xslt
info: Copying xUnit1.xslt
info: Copying xunit.abstractions.dll
info: Copying xunit.console.exe
info: Copying xunit.console.exe.config
info: Copying xunit.console.x86.exe
info: Copying xunit.console.x86.exe.config
info: Copying xunit.runner.reporters.desktop.dll
info: Copying xunit.runner.utility.desktop.dll
info: Copying xunit.execution.desktop.dll
info: Exec
info:   program: /Users/davidobando/github/aspnet/Performance/test/Microbenchmarks.Tests/obj/testPublish-dnx451/xunit.console.exe
info:   commandline: Microbenchmarks.Tests.dll  -parallel none
info:   workingdir: test/Microbenchmarks.Tests/obj/testPublish-dnx451
xUnit.net Console Runner (64-bit .NET 4.0.30319.17020)
  Discovering: Microbenchmarks.Tests
  Discovered:  Microbenchmarks.Tests
  Starting:    Microbenchmarks.Tests
CalibrationTests.Calibration_100ms (Variation=Default)
    Warmup Iterations: 1
    Collection Iterations: 100
    Time Elapsed (95th Percentile): 103ms
    Memory Delta (95th Percentile): 0
Can't connect to the specified datasource (localdb)\mssqllocaldb
CalibrationTests.Calibration_100ms_controlled (Variation=Default)
    Warmup Iterations: 1
    Collection Iterations: 100
    Time Elapsed (95th Percentile): 104ms
    Memory Delta (95th Percentile): 0
HostingTests.MainToConfigureOverhead (Variation=Kestrel)
    Warmup Iterations: 1
    Collection Iterations: 100
    Time Elapsed (95th Percentile): 1ms
    Memory Delta (95th Percentile): 16920
    MainToConfigureOverhead [Variation: WebListener] [Stage: Warmup] [Iteration: 1/1] [FAIL]
      System.TypeInitializationException : The type initializer for 'HttpApi' threw an exception.
      ---- System.DllNotFoundException : httpapi.dll
      Stack Trace:
        <filename unknown>(0,0): at Microsoft.Net.Http.Server.WebListener..ctor (ILoggerFactory factory)
        <filename unknown>(0,0): at Microsoft.AspNetCore.Server.WebListener.ServerFactory.CreateServer (IConfiguration configuration)
        <filename unknown>(0,0): at Microbenchmarks.Tests.HostingTests+TestServerLoader+TestServerFactory.CreateServer (IConfiguration configuration)
        <filename unknown>(0,0): at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureServer ()
        <filename unknown>(0,0): at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication ()
        <filename unknown>(0,0): at Microsoft.AspNetCore.Hosting.Internal.WebHost.Initialize ()
        <filename unknown>(0,0): at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build ()
        <filename unknown>(0,0): at Microbenchmarks.Tests.HostingTests.MainToConfigureOverhead (System.String variationServer)
           at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
        <filename unknown>(0,0): at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture)
        ----- Inner Stack Trace -----
           at (wrapper managed-to-native) Microsoft.Net.Http.Server.UnsafeNclNativeMethods+HttpApi:HttpInitialize (Microsoft.Net.Http.Server.UnsafeNclNativeMethods/HttpApi/HTTPAPI_VERSION,uint,void*)
        <filename unknown>(0,0): at Microsoft.Net.Http.Server.UnsafeNclNativeMethods+HttpApi.InitHttpApi (UInt16 majorVersion, UInt16 minorVersion)
        <filename unknown>(0,0): at Microsoft.Net.Http.Server.UnsafeNclNativeMethods+HttpApi..cctor ()
```

**Comment by moozzyk** [comment]
I think WebListener is Windows specific and you should not run this variation when not on Windows...

**Comment by DavidObando** [comment]
@moozzyk do you know who I can reach out to in order to learn how to not run variations on specific operating systems?

/cc @SajayAntony

**Comment by moozzyk** [comment]
@DavidObando - I don't know this repo but maybe the attributes we use in other repos would work? (e.g. https://github.com/aspnet/dnx/blob/166d3e561f1c0094c72952765c00bc33e0a4c2d7/test/Microsoft.Dnx.Runtime.Tests/DependencyManagement/GacDependencyResolverFacts.cs#L18-L19)

**Comment by SajayAntony** [comment]
@troydai @muratg - this test should not show up on non-windows platforms. I don't know if there is a property to determine this.

**Comment by troydai** [comment]
@SajayAntony The `PlatformServices` can help to determine the current OS. An also @moozzyk pointed out a common approach to skip tests on a specific platform.

**Comment by NTaylorMullen** [comment]
If this applies I'd definitely using something like the os skip condition that @moozzyk linked to.

**Comment by DavidObando** [comment]
Solved with fff5171.
